pkg/resources/task.go

```diff
@@ -284,7 +285,7 @@ func ReadTask(d *schema.ResourceData, meta interface{}) error {
 }
 
 // CreateTask implements schema.CreateFunc.
-func CreateTask(d *schema.ResourceData, meta interface{}) error {
+func CreateTask(d *schema.ResourceData, meta interface{}) (returnedErr error) {
```

Comment:
The test I mentioned:
```
func TestNamedReturn(t *testing.T) {
	g := func(e error) error {
		if e != nil {
			return e
		}
		return nil
	}

	def := func(e error) error {
		if e != nil {
			return e
		}
		return nil
	}

	f := func(mainErr error, anotherErr error, deferredErr error) (returnedErr error) {
		defer func() {
			returnedErr = errors.Join(returnedErr, def(deferredErr))
		}()
		if mainErr != nil {
			return mainErr
		}
		return g(anotherErr)
	}

	me := errors.New("main function error")
	ae := errors.New("another function error")
	de := errors.New("deferred function error")

	_, _ = ae, de

	t.Run("no errors", func(t *testing.T) {
		returnedErr := f(nil, nil, nil)

		require.Nil(t, returnedErr)
	})

	t.Run("error in main func, no errors in deferred", func(t *testing.T) {
		returnedErr := f(me, nil, nil)

		// they are not equal, because the packaging takes place in defer
		require.NotEqual(t, returnedErr, me)
		require.Contains(t, returnedErr.Error(), me.Error())
		require.NotContains(t, returnedErr.Error(), ae.Error())
		require.NotContains(t, returnedErr.Error(), de.Error())
	})

	t.Run("error in another func, no errors in deferred", func(t *testing.T) {
		returnedErr := f(nil, ae, nil)

		// they are not equal, because the packaging takes place in defer
		require.NotEqual(t, returnedErr, ae)
		require.NotContains(t, returnedErr.Error(), me.Error())
		require.Contains(t, returnedErr.Error(), ae.Error())
		require.NotContains(t, returnedErr.Error(), de.Error())
	})

	t.Run("no errors in main and in another, error in deferred", func(t *testing.T) {
		returnedErr := f(nil, nil, de)

		require.NotEqual(t, returnedErr, de)
		require.NotContains(t, returnedErr.Error(), me.Error())
		require.NotContains(t, returnedErr.Error(), ae.Error())
		require.Contains(t, returnedErr.Error(), de.Error())
	})

	t.Run("error in main func, error in deferred", func(t *testing.T) {
		returnedErr := f(me, nil, de)

		require.NotEqual(t, returnedErr, me)
		require.NotEqual(t, returnedErr, de)
		require.Contains(t, returnedErr.Error(), me.Error())
		require.NotContains(t, returnedErr.Error(), ae.Error())
		require.Contains(t, returnedErr.Error(), de.Error())
	})

	t.Run("error in another func, error in deferred", func(t *testing.T) {
		returnedErr := f(nil, ae, de)

		require.NotEqual(t, returnedErr, ae)
		require.NotEqual(t, returnedErr, de)
		require.NotContains(t, returnedErr.Error(), me.Error())
		require.Contains(t, returnedErr.Error(), ae.Error())
		require.Contains(t, returnedErr.Error(), de.Error())
	})
}

```

Reply:
Yes, let's do it like this at least for now, but add this as a consideration for the follow-up ticket (I mean handling the resume errors).